internal/provider/kubernetes/kubernetes.go

```diff
@@ -110,7 +111,7 @@ func New(cfg *rest.Config, svr *ec.Server, resources *message.ProviderResources)
 	// Emit elected & continue with deployment of infra resources
 	go func() {
 		<-mgr.Elected()
-		close(svr.Elected)
```

Comment:
logic is changing here ...

Reply:
a channel can only be closed once

Reply:
sg, do we have a leader election test ?
cc @alexwo

Reply:
tracked in https://github.com/envoyproxy/gateway/issues/4420